search-improvements.rst

```diff
@@ -0,0 +1,189 @@
+  where the `Mozilla` word is present, but `Firefox` word is not present.
+  For more information, you can look at `Simple Query String Syntax`_ documentation.
+
+- `Case Insensitive Search`_: The search is now case insensitive. So if you search for `Foo`,
+  you will get all the documentations which have one of either `Foo`, `FOO`, `foo` or `fOO`.
+
+- `Improved Result order`_: The result order is improved dramatically. Therefore if you Search
+  with `Foo Bar`, you will first see the documentations which have both `Foo Bar`, then
+  you will see other documentations which have either `Foo` or `Bar`
+
+- `Auto Removing from Search Index`_: In the past, if any page got removed from documentations,
+  the page was still available in documentations. But from now on, the page will be removed
+  automatically from the search index as soon as its no longer in the documentation.
+
+- `Zero Downtime Indexing`_: As search is a important part of documentation, there will be no
+  downtime while we reindex our Elasticsearch Index. So the search will be much more reliable
+  than before.
+
+
+Performance Improvements
+^^^^^^^^^^^^^^^^^^^^^^^^
+Performance is always improtant for search. Because of our search was fast enough,
+we did not focus to improve the performance. As we have rewritten all the codes with
+new search backend engine, we got 4x improved performance as well.
+
+There are 4 kind of search functionality in Read The Docs. The improvements are as following:
+
+- `File Search`_: Search accross all the projects of Read The Docs.
+
+  - Before: 1461 ms
+  - After: 200 ms
+
+- `Projects Search`_: Search for projects hosted in Read The Docs.
+
+  - Before: 455 ms
+  - After: 162 ms
+
+- `Doc Search API`_: The Public API which is used to perform search in documentation.
+
+  - Before: 746 ms
+  - After: 173 ms
+
+- `Project Doc Search`_: Search inside a project documentations from readthedocs.org domain.
+
+  - Before: 750 ms
+  - After: 270 ms
+
+
+Code Improvements
+^^^^^^^^^^^^^^^^^
+Code quality is very much important in development world, specially in open source.
+As I have rewritten the search functionality from scratch, the code quality
+is improved in many ways like test coverage and documentations. So its easy for
+any contributor to start working on the search functionality
+
+Contributor Wanted
+^^^^^^^^^^^^^^^^^^
+As Read The Docs is an open source project backed by a small team of developers,
+most of them are busy to keep things up and running only. Therefore, its quite
+hard for them to take time to implement new features. If you know some bit of
+Django or Python and Elasticsearch, you can contribute into the search functionality
+of Read The Docs. If you need any support to start contributing, you can knock me or any
+member of  Read The Docs team. You can find all of us at `#readthedocs` freenode
+IRC channel or `readthedocs gitter`_ channel. I am `safwan` at IRC and `@safwanrahman`
+at gitter.
+
+Conclusion
+^^^^^^^^^^
```

Comment:
One thing missing from the conclusion that I think should be in there is about how much you learned as part of GSoC?

Reply:
I can include that, but I think this blog post is more focused on the search improvement of read the docs. so I am not sure if it should be included